resources/ext.wikibase.facetedsearch.js

```diff
@@ -75,19 +79,32 @@ function onInstancesClick( event, instanceId ) {
 /**
  * Handles the input event for a list facet.
  *
- * @param {HTMLDivElement} facet - The facet element.
- * @param {string} propertyId - The ID of the property on the facet.
+ * @param {HTMLDivElement} facet
+ * @param {string} propertyId
+ * @param {?string} mode
  */
-function onListFacetInput( facet, propertyId ) {
-	const newQueries = getListFacetQuerySegments( facet, propertyId );
+function onListFacetInput( facet, propertyId, mode ) {
+	mode = mode || getListFacetQueryMode( facet );
+	const newQueries = getListFacetQuerySegments( facet, propertyId, mode );
 	submitSearchForm( buildQueryString( specialSearchInput.value, newQueries, propertyId ) );
 }
 
+/**
+ * Determines the query mode for a list facet based on the selected toggle button.
+ *
+ * @param {HTMLDivElement} facet
+ * @return {string}
+ */
+function getListFacetQueryMode( facet ) {
+	const selectedButton = facet.querySelector( '.wikibase-faceted-search__facet-toggle > .cdx-button--action-progressive' );
+	return selectedButton ? selectedButton.value : 'AND';
```

Comment:
Normally there should always be a selected button, right? In other words, this check is to deal with unexpected error states.

If not, and we can regularly have the state, then defauling to AND is likely incorrect behavior.

Reply:
Yes normally there are always a selected button, this is used to deal with unexpected state as you mentioned